### ISSUE TITLE
HDMI audio codec crashes when no monitor is plugged in

### ISSUE THREAD

**Issue by leigh-johnson** [bitsy-ai/printnanny-os#212]
## **Describe the bug**
I'm seeing the following dmesg in the v0.6.0 release candidate. 
```
  284.366578] hdmi-audio-codec hdmi-audio-codec.0.auto: ASoC: error at snd_soc_dai_startup on i2s-hifi: -19
[  284.366729]  MAI: soc_pcm_open() failed (-19)
[  284.367514] hdmi-audio-codec hdmi-audio-codec.1.auto: ASoC: error at snd_soc_dai_startup on i2s-hifi: -19
[  284.367588]  MAI: soc_pcm_open() failed (-19)
```

## **To Reproduce**
What are the steps to reproduce the issue?

1. Without an HDMI monitor plugged in, start a gstreamer pipeline with `mp4mux` element, which probes the HDMI audio codec 

## Issue.txt

```
-----------------------
Build Info:  |
-----------------------
IMAGE_NAME = printnanny-release-image-raspberrypi4-64-20230120182810
DISTRO_PRETTY = PrintNanny Linux 0.6.0 (Feldspar)
VARIANT_NAME = PrintNanny OS Core
VARIANT_ID = core
DISTRO = printnanny
DISTRO_NAME = PrintNanny Linux
DISTRO_VERSION = 0.6.0
DISTRO_CODENAME = Feldspar
ID_LIKE = BitsyLinux
HOME_URL = https://printnanny.ai
BUG_REPORT_URL = https://github.com/bitsy-ai/printnanny-os/issues
YOCTO_VERSION = 4.0.2
YOCTO_CODENAME = Langdale
MACHINE = raspberrypi4-64
TUNE_PKGARCH = cortexa72
-----------------------
Layer Revisions:      |
-----------------------
meta              = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-poky         = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-yocto-bsp    = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-raspberrypi  = libcamera-v0.0.3:104a8b9290ab4da92c8f693e2fdc42a6396975d2
meta-oe           = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-python       = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-multimedia   = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-bitsy        = main:907e754b35e50473b8707a8f3e69705ce07257a1
meta-neural-network = master:fed68f0a659c9b6431985c156141dc28dc4c2275
meta-printnanny   = main:907e754b35e50473b8707a8f3e69705ce07257a1
meta-networking   = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-filesystems  = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-initramfs    = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-webserver    = langdale:c5668905a6d8a78fb72c2cbf8b20e91e686ceb86
meta-swupdate     = master:5e611f44db1b9333342ed61d77a3c3ed7be69c2b
workspace         = langdale:a3e3b740e140d036122f7b11e2ac452bda548444
meta-microcontroller = master:9f396e595b2d639be7f2b494f4a61e48d9495058
```

**Comment by leigh-johnson** [comment]
Kernel version:
```
pn-v0-6:/home/pi# uname -a
Linux pn-v0-6 5.15.56-v8 #1 SMP PREEMPT Fri Jul 22 13:23:23 UTC 2022 aarch64 GNU/Linux
```

Firmware version:
```
pn-v0-6:/home/pi# vcgencmd version
Aug 26 2022 14:04:10 
Copyright (c) 2012 Broadcom
version 102f1e848393c2112206fadffaaf86db04e98326 (clean) (release) (start_x)
```

Error source + mailing list. Looks like this is specifically happening when Gstreamer probes for available audio codecs, but since no display is plugged in, HDMI codec hasn't been initialized. 
> It pre-dates my time working on the vc4 driver so I'm not really sure
what this is supposed to prevent, but my guess is that it's there to
avoid someone using the audio card before we have a display detected and
connected, and its capabilities known (the first and more obvious one
being does it support audio in the first place).

https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/gpu/drm/vc4/vc4_hdmi.c#n1083
https://lore.kernel.org/all/45e55e16-6bb7-5d82-1031-db5deaee68ab@i2se.com/T/

**Comment by leigh-johnson** [comment]
Pulling in a Raspberry Pi firmware patch that reworks v4l2 memory management. Minimum kernel version required is 5.15.67, but I'm going to try out [5.15.89](https://github.com/raspberrypi/firmware/commit/2578acb89b6b40c483db537c4fd2eef593543fb8)
https://github.com/raspberrypi/firmware/commit/ab68242d6131d078b757a6962d7b22a3da79d3d9

**Comment by leigh-johnson** [comment]
:tada: Fixed using kernel 5.15.87 and the `20230106` Rasperry Pi firmware release
```
Linux pn-v0-6 5.15.87-v8 #1 SMP PREEMPT Fri Jan 13 16:51:07 UTC 2023 aarch64 GNU/Linux
```
